// gatsby-plugin-transition-link/components/PageWrapper/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

exports.__esModule = true;
exports.PageWrapper = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _inheritsLoose2 = _interopRequireDefault(require("@babel/runtime/helpers/inheritsLoose"));

var _react = _interopRequireWildcard(require("react"));

var _createTransitionContext = require("../../context/createTransitionContext");

require("./style.css");

var _jsxFileName = "/Users/tylerbarnes/Documents/gatsby/transitionlink-workspace/gatsby-plugin-transition-link/src/components/PageWrapper/index.js";

var PageWrapper =
/*#__PURE__*/
function (_Component) {
  (0, _inheritsLoose2.default)(PageWrapper, _Component);

  function PageWrapper() {
    return _Component.apply(this, arguments) || this;
  }

  var _proto = PageWrapper.prototype;

  _proto.shouldComponentUpdate = function shouldComponentUpdate(previousProps) {
    return previousProps.transitionStatus !== this.props.transitionStatus;
  };

  _proto.render = function render() {
    var _this$props = this.props,
        children = _this$props.children,
        transitionState = _this$props.transitionState,
        transitionStatus = _this$props.transitionStatus,
        exitProps = _this$props.exitProps,
        entryProps = _this$props.entryProps,
        forwardedRef = _this$props.forwardedRef;
    var mount = transitionStatus === "entering" || transitionStatus === "entered";
    var exitZindex = exitProps.zIndex || 0;
    var entryZindex = entryProps.zIndex || 1;
    return _react.default.createElement("div", {
      className: "tl-wrapper " + (mount ? "tl-wrapper--mount" : "tl-wrapper--unmount") + " tl-wrapper-status--" + transitionStatus,
      style: {
        zIndex: mount ? entryZindex : exitZindex
      },
      __source: {
        fileName: _jsxFileName,
        lineNumber: 28
      },
      __self: this
    }, _react.default.createElement(_createTransitionContext.PublicProvider, {
      value: (0, _extends2.default)({}, transitionState),
      __source: {
        fileName: _jsxFileName,
        lineNumber: 36
      },
      __self: this
    }, children));
  };

  return PageWrapper;
}(_react.Component);

exports.PageWrapper = PageWrapper;
